src/Host.cpp

```diff
@@ -3537,6 +3537,35 @@ bool Host::setLabelOnLeave(const QString& name, const int func)
     return false;
 }
 
+std::pair<bool, QString> Host::setMovie(const QString& name, const QString& moviePath)
+{
+    if (!mpConsole) {
+        return {false, QString()};
+    }
+
+    auto pL = mpConsole->mLabelMap.value(name);
+    if (!pL) {
+        return {false, qsl("label '%1' does not exist").arg(name)};
+    }
+
+    auto myMovie = pL->movie();
+    if (!myMovie) {
+        myMovie = new QMovie();
+        myMovie->setCacheMode(QMovie::CacheAll);
+    }
+
+    myMovie->setFileName(moviePath);
+
+    if (!myMovie->isValid()) {
+        return {false, qsl("no valid movie found at '%1'").arg(moviePath)};
+    }
+    myMovie->stop();
+    pL->setMovie(myMovie);
```

Comment:
> Sets the label contents to movie. Any previous content is cleared. The label does NOT take ownership of the movie.

I think this creates a memory leak, doesn't it? if you create mudlet profile with a lot of gifs and then close it, does all the memory go away?

Reply:
Arg. Yes, it does.
I found also another memory leak.
If echoing to a label for example it will also set qlabel->movie() to nullptr and then if setting a new movie it won't delete the old one. 
Will solve both issues.